lib/CL/pocl_raw_ptr_set.h

```diff
@@ -0,0 +1,106 @@
+/* Provides container for mapping VM and BDA addresses to pocl_raw_ptr
+   objects.
+
+   Copyright (c) 2025 Henry Linjam�ki / Intel Finland Oy
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to
+   deal in the Software without restriction, including without limitation the
+   rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+   sell copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in
+   all copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+   IN THE SOFTWARE.
+*/
+#ifndef POCL_RAW_PTR_SET_H
+#define POCL_RAW_PTR_SET_H
+
+#include "pocl_raw_ptr.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+  typedef struct pocl_raw_ptr_set pocl_raw_ptr_set;
+
+  /** Creates new empty pocl_raw_ptr_set.
+   */
+  POCL_EXPORT
```

Comment:
Can you not indent due to the extern "C" block? I know clang-format wants to indent here and I couldn't find a way to disable it.